### ISSUE TITLE
"Allowed file extensions" -> "Limit file types uploading to" incorrectly applying 

### ISSUE THREAD

**Issue by quazarjim** [projectsend/projectsend#563]
In Security -> Allowed file extensions, "Limit file types uploading to" is set to "Clients Only".

I am logged in with an account that is a "System Administrator".

I try to upload a file that is not present in the "Allowed file extensions" list.  It disallows it, but this doesn't make sense to me,  To my understanding, the Allowed File extensions limit should not apply to me, since I am not a client.

After adding the file extension to the list, I can then upload a file.

**Comment by lenamtl** [comment]
This will required custom code to allow extension per role.

**Comment by mschop** [comment]
@lenamtl seems like a bug to me. Why should custom code be needed here?

**Comment by WowHesTall** [comment]
Hey there @quazarjim - if you edit **process-upload.php** to this:

```
// Validate file has an acceptable extension
if ( false === CAN_UPLOAD_ANY_FILE_TYPE ) {
	$fileExt = pathinfo($fileName, PATHINFO_EXTENSION);
	$allowedExt = explode(',', $options_values['allowed_file_types'] );
	if (!in_array($fileExt, $allowedExt)) {
		die('{"jsonrpc" : "2.0", "error" : {"code": 104, "message": "Invalid Extension."}, "id" : "id"}');
	};
}
```

Should all work fine now.

**Comment by ignacionelson** [comment]
Solved in r1265, thanks @WowHesTall

**Comment by SonicDiscovery** [comment]
I am getting this error when testing a user profile for uploading an mp3. In admin, the file uploads fine.
MP3: Invalid Extension

Any ideas to fix are appreciated, thank you

**Comment by cburlacu** [comment]
@SonicDiscovery It seems that it is already fixed in ec6ef20c5034504697fb266921ae6da73d2d545a
